File: CompetitiveProgramming/COCI/COCI '13 Contest 2 #3 Slom/2021-03-11/Slom.java
import java.util.*;
import java.io.*;

public class Slom
{
	static int n;
	static String s;
	static String original;
	
	static void calc()
	{
		if(s.length()%2==0)
		{
			for(int i = s.length()-3; i >= 0; i--)
				if(i%2!=0)
					s = s.substring(0, i) + s.substring(i+1) + s.charAt(i);
		}
		else
		{
			for(int i = s.length()-2; i >= 0; i--)
				if(i%2!=0)
					s = s.substring(0, i) + s.substring(i+1) + s.charAt(i);
		}
	}
	
	public static void main (String [] args) throws IOException
	{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		n = Integer.parseInt(br.readLine());
		s = br.readLine();
		original = s;
		String [] data = new String [10000];
		int ind = 0;
		//n %= (int)(s.substring(1).chars().distinct().count());
		
		int period = n+1;
		data[ind++] = s;
		for(int i = 0; i < n; i++)
		{
			calc();
			if(s.equals(original))
			{
				period = i+1;
				break;
			}
			data[ind++] = s;
		}
		System.out.println(data[n%period]);
	}
}

File: CompetitiveProgramming/COCI/COCI '13 Contest 2 #3 Slom/2021-03-12/Slom.cpp
#include <bits/stdc++.h>
using namespace std;

int n;
string s;
string original;

void calc()
{
	if(s.length()%2==0)
	{
		for(int i = s.length()-3; i >= 0; i--)
			if(i%2!=0)
			{
				s = s.substr(0, i) + s.substr(i+1, s.length()-1-i) + s[i];
	//cout << i << ":" << n << endl << s << endl; ///////////////////
			}
	}
	else
	{
		for(int i = s.length()-2; i >= 0; i--)
			if(i%2!=0)
				s = s.substr(0, i) + s.substr(i+1, s.length()-i-1) + s[i];
	}
}

int main()
{
	cin.sync_with_stdio(0);
	cin.tie(0);
	
	cin >> n >> s;
	//cout << n << endl << s << endl; ///////////////////
	original = s;
	string data [10000];
	int ind = 0;
	
	int period = n+1;
	data[ind++] = s;
	for(int i = 0; i < n; i++)
	{
		calc();
		if(s.compare(original) == 0)
		{
			period = i+1;
			break;
		}
		data[ind++] = s;
	}
	cout << data[n%period] << endl;
	return 0;
}